app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/acquisitiondetails/AcquisitionDetailsController.scala

```diff
@@ -408,35 +418,37 @@ class AcquisitionDetailsController @Inject() (
             case (assetType, wasUkResident) =>
               withAcquisitionDate(answers) { acquisitionDate =>
-                rebasingCutOffDate(acquisitionDate, assetType, wasUkResident).fold[Future[Result]](
-                  Redirect(routes.AcquisitionDetailsController.checkYourAnswers())
-                ) { rebaseDate =>
-                  commonSubmitBehaviour(fillingOutReturn, answers)(
-                    form = rebasedAcquisitionPriceForm
-                  )(
-                    page = {
-                      case (form, backLink) =>
-                        val p = form.copy(errors =
-                          form.errors.map(_.copy(args = Seq(LocalDateUtils.govDisplayFormat(rebaseDate))))
-                        )
-                        rebasedAcquisitionPricePage(p, backLink, rebaseDate)
-                    }
-                  )(
-                    requiredPreviousAnswer = _.fold(
-                      _.acquisitionPrice,
-                      c => Some(c.acquisitionPrice)
-                    ),
-                    redirectToIfNoRequiredPreviousAnswer = routes.AcquisitionDetailsController.acquisitionPrice()
-                  )(
-                    updateAnswers = {
-                      case (p, answers) =>
-                        answers.fold(
-                          _.copy(rebasedAcquisitionPrice = Some(AmountInPence.fromPounds(p))),
-                          _.copy(rebasedAcquisitionPrice = Some(AmountInPence.fromPounds(p)))
-                        )
-                    }
-                  )
-                }
+                rebasingEligabilityUtil
+                  .rebasingCutOffDate(acquisitionDate, assetType, wasUkResident)
+                  .fold[Future[Result]](
+                    Redirect(routes.AcquisitionDetailsController.checkYourAnswers())
+                  ) { rebaseDate =>
+                    commonSubmitBehaviour(fillingOutReturn, answers)(
+                      form = rebasedAcquisitionPriceForm
+                    )(
+                      page = {
+                        case (form, backLink) =>
+                          val p = form.copy(errors = form.errors
+                            .map(_.copy(args = Seq(LocalDateUtils.govDisplayFormat(rebaseDate.minusDays(1)))))
+                          )
+                          rebasedAcquisitionPricePage(p, backLink, rebaseDate.minusDays(1))
+                      }
+                    )(
+                      requiredPreviousAnswer = _.fold(
+                        _.acquisitionDate,
+                        c => Some(c.acquisitionDate)
+                      ),
+                      redirectToIfNoRequiredPreviousAnswer = routes.AcquisitionDetailsController.acquisitionDate()
+                    )(
+                      updateAnswers = {
+                        case (p, answers) =>
+                          answers.fold(
+                            _.copy(rebasedAcquisitionPrice = Some(AmountInPence.fromPounds(p))),
```

Comment:
for uk residents who we ask this question to should we not also set `shouldUseRebase` to `true` and set the `acquisitionPrice` field as well? It'd be a bit safer to do it at this point rather than in the cya endpoint so we can se it all happening at the same time in one place - but also it looks like you're doing `.getOrElse(false)` for `shouldUseRebase` which would set it to the wrong value in this scenario.